### DISCUSSION TITLE
spotify: cannot login in browser

### DISCUSSION THREAD

**Discussion by nsauter** [netblue30/firejail#5912] [Q&A] [answered]
Hi!

I want to start my spotify client in a firejail. I managed to do that and if i specify --username and --password (spotify arguments) i can login right away and start streaming. If i dont do that spotify opens and wants me to login with a browser. But of course no browser is opened, because the jail is not allowed to open a popup/browser i guess?

How can i manage to login into spotify without the Browser redirect?

**Comment by ghost** [comment, marked as answer]
There are quite a few spotify clients out there. Can you tell us which specific application you're trying to get working?

> How can i manage to login into spotify without the Browser redirect?

Didn't you manage to do that by specifying --username and --password flags to spotify? Or is that only working without Firejail? A bit confused here.

> But of course no browser is opened, because the jail is not allowed to open a popup/browser i guess?

Spot on. The current `spotify.profile` doesn't allow external web browser communications from inside its sandbox. But you can try adding support for `Firefox` like some profiles already do:

```sh
$ cat ~/.config/firejail/spotify.local
# The lines below are needed to find the default Firefox profile name, to allow
# opening links in an existing instance of Firefox (note that it still fails if
# there isn't a Firefox instance running with the default profile; see #5352)
noblacklist ${HOME}/.mozilla
whitelist ${HOME}/.mozilla/firefox/profiles.ini

dbus-user filter
# allow D-Bus communication with firefox for opening links
dbus-user.talk org.mozilla.*
# allow MPRIS (I'm not sure on this one)
dbus-user.own org.mpris.MediaPlayer2.*
# blocking the system-bus should be fine for most apps
dbus-system none
```

You'll have to experiment a bit here.
